apiserver/Makefile

```diff
@@ -32,6 +32,14 @@ jupyter_notebook_config.py:
 	    -e "s,@keyfile@,$(GSA_KEY_FILE),g" \
             < jupyter_notebook_config.py.in > $@
 
+hailjwt: ../hailjwt/setup.py $(find ../hailjwt/hailjwt -iname \*.py)
+	-mkdir hailjwt
+	cp ../hailjwt/setup.py hailjwt/
+	rsync -a --prune-empty-dirs --include '*/' --include '*.py' --exclude '*' ../hailjwt/hailjwt/ hailjwt/
```

Comment:
I like the rsync, by my rsync-fu is weak.  How does --prune-empty-dirs and --include '*/' work?  Isn't that redundant?  I thought we should only be copying for docker to avoid the large context issue.

Reply:
In trying to explain this I settled on a better solution:
```Makefile
hailjwt: ../hailjwt/setup.py $(shell find ../hailjwt/hailjwt -iname \*.py)
	cd ../ && rsync --relative $(?:../hailjwt/%=hailjwt/./%) apiserver/hailjwt
	touch hailjwt
```

`hailjwt` is a target whose timestamp will represent the last time we sync'ed it with ../hailjwt. It depends on setup.py as well as any py files in the python package.

We want to copy the out of date dependencies (`$?`) preserving their directory structure. `cp` cannot do this (on bsd). `rsync --relative` preserves directory structure of inputs. `rsync` refuses to use `--relative` when you have `../` in an input path. `--relative`, by default replicates the entire input path at the destination, unless it sees a `/./`, which it instead uses as the root of the newly copied path. To accommodate `rsync`, we go up one dir and use `$(?:../hailjwt/%=hailjwt/./%)` to fix up the paths for rsync. Example follows.

```
(hail) dking@wmb16-359 # rm -rf hailjwt && make hailjwt
cd ../ && rsync --relative hailjwt/./setup.py hailjwt/./hailjwt/__init__.py hailjwt/./hailjwt/hailjwt.py apiserver/hailjwt
touch hailjwt
(hail) dking@wmb16-359 #  make hailjwt                 
make: `hailjwt' is up to date.
(hail) dking@wmb16-359 # touch ../hailjwt/hailjwt/__init__.py
(hail) dking@wmb16-359 #  make hailjwt
cd ../ && rsync --relative hailjwt/./hailjwt/__init__.py apiserver/hailjwt
touch hailjwt
(hail) dking@wmb16-359 #  make hailjwt
make: `hailjwt' is up to date.
(hail) dking@wmb16-359 # ls hailjwt
hailjwt  setup.py
(hail) dking@wmb16-359 # ls hailjwt/hailjwt 
__init__.py hailjwt.py
```